software/base/src/main/java/org/apache/brooklyn/entity/software/base/VanillaWindowsProcessWinRmDriver.java

```diff
@@ -49,15 +50,29 @@ public void start() {
 
     @Override
     public void install() {
+        // TODO: At some point in the future, this should probably be refactored to get the name of the machine in WinRmMachineLocation and set it as the hostname sensor
+        String hostname = null;
+        if (entity.getConfig(VanillaWindowsProcess.INSTALL_REBOOT_REQUIRED)) {
+            WinRmExecuteHelper checkHostnameTask = newScript("Checking hostname")
+                    .setCommand("hostname")
+                    .failOnNonZeroResultCode()
+                    .gatherOutput();
+            if (checkHostnameTask.execute() != 0) {
+                LOG.warn("Cannot determine hostname for entity {} machine {}", getEntity(), getLocation());
+            }
+            hostname = Strings.trimEnd(checkHostnameTask.getResultStdout());
```

Comment:
Should we really be extracting hostname from stdout if the command failed to execute (i.e. if we did the above log statement)? Presumably we'll get either an empty string or some error message. What are the implications of subsequently passing that to `rebootAndWait(hostname)`?